lib/src/iter.rs

```diff
@@ -61,36 +60,38 @@ impl<'r> Iterator for HeadRefsToIssuesIter<'r>
 /// until an initial issue message is encountered, inclusively.
 ///
 pub struct IssueMessagesIter<'r> {
-    inner: FirstParentIter<'r>,
+    inner: git2::Revwalk<'r>,
     repo: &'r Repository,
 }
 
 impl<'r> IssueMessagesIter<'r> {
-    pub fn new<'a>(commit: Commit<'a>, repo: &'a Repository) -> IssueMessagesIter<'a> {
-        IssueMessagesIter {
-            inner: FirstParentIter::new(commit),
-            repo: repo,
+    pub fn new<'a>(repo: &'a Repository, commit: Commit<'a>) -> Result<IssueMessagesIter<'a>> {
+        let revwalk = repo.first_parent_revwalk(commit.id())?;
+        Ok(IssueMessagesIter { inner: revwalk, repo: repo })
```

Comment:
Why not 

```rust
repo
    .first_parent_revwalk(commit.id())
    .map(|rw| IssueMessagesIter {
        inner: rw,
        repo: repo
    })
```

(Although I'm not sure whether the compiler is smart enough to figure this out...)